docs/docs/contributing/tools/ide.md

```diff
@@ -0,0 +1,38 @@
+---
+layout: doc-page
+title: Using an IDE
+---
+
+You can use either Metals (VS Code, vim) or IntelliJ IDEA as described on the
+[IDE Support](../../usage/ide-support.md) page to work on the Scala 3 codebase. There are however
+a few additional considerations to take into account.
+
+
+## Bootstrapping Projects
+
+The sbt build for dotty implements bootstrapping within the same build, so each component has
+two projects in the sbt build:
+
+```
+sbt:scala3> projects
+...
+[info] 	   scala3-compiler
+[info] 	   scala3-compiler-bootstrapped
+...
+```
+
+When working in an IDE these duplicated projects can be confusing or cause issues.
+
+For Metals we recommend using
+[sbt as BSP server](https://scalameta.org/metals/blog/2020/11/06/sbt-BSP-support.html) instead
+of bloop, then the `-bootstrapped` projects are hidden.
+
+In IntelliJ IDEA, we recommend to import the dotty codebase through BSP as described on
+the [IDE Support page](../../usage/ide-support.md), then the `-bootstrapped` projects are hidden.
+
+
+## Scala Version warning in Metals
+
+When using VS Code, Metals will likely show a warning that the Scala version (`3.0.0-[...]-NIGHTLY`)
+is not supported. The reason is that the dotty repository often uses a nightly build as reference
+compiler. The IDE experience is therefore going to be limited.
```

Comment:
I'm not sure how limited the experience is going to be - @tgodzik maybe you can clarify?

Reply:
Mostly it will work on a single file basis. Nothing project wide. It is however quite easy to publish Metals for the nightly versions. I wonder if it would make sense to have a CI job that would publish Metals for the nightly versions. It would really require `++<nightly-version> mtags/publish` and some small work around the sbt build to accept the nightly. We could have an env variable that would be added to the scalaVersions and that should work well enough. An we would also need to make Metals alway try to download the nigthlies.

Alternatively, we could at least add some instructions on how to publish Metals locally:

- add nightly version to `scala3Versions` in build.sbt
- run `+publishLocal` if the current snapshot was not yet published locally otherwise it can just be published for the single version.
- change the Metals version to the snapshot and reload.

It could even be a staging Metals repo that we would work from and some special org handling in Metals.

Reply:
What is the technical reason that Metals needs to be fully cross-versioned? IIUC, in Scala 2 there's the semanticdb compiler plugin that requires it, but dotc emits semanticdb itself.

Reply:
We depend on specific version of the Scala 3 compiler, I was thinking about maybe extracting an API for Metals to use, but haven't had the chance to take a look at it.

Reply:
Thanks for the details @tgodzik!

Reply:
(Likely, the usage of nightlies as reference compiler will be less common now that things stabilize. Currently the reference version is 3.0.0-RC1).